framework.py

```diff
@@ -9,16 +9,150 @@
 # not expressly granted therein are reserved by Shotgun Software Inc.
 
 import sgtk
+from sgtk.util import filesystem
+
+import threading
+import datetime
+import os
+import time
 
 class ShotgunUtilsFramework(sgtk.platform.Framework):
     
+    # A list of file names which should never be deleted when cleaning up old
+    # cached data.
+    _ALWAYS_KEEP_CACHED_FILES = [
```

Comment:
i know that this is cache data, but even so i am a bit paranoid about deleting. I would prefer if we had an opt-in approach rather than opt out. Instead of saying "delete everything except the schema files" we should say "delete thumbs and caches" - to avoid deleting something unintentionally. If we add some module to the sg utils in the future, it's easy to forget to add it to exclude list and we won't notice the behaviour until much later and the deletion kicks in. Also, maybe i am overthinking but there could be forks of sgutils where clients cache data etc. Just seems safer to be explicit in this case.